src/template/L1PortalExecuteL2Call.sol

```diff
@@ -95,8 +86,12 @@ contract L1PortalExecuteL2Call is SimpleTaskBase {
 
     /// @notice Build the portal deposit action. WARNING: State changes here are reverted after capture.
     function _build(address) internal override {
-        // Record the L1 portal call with value for action extraction.
-        IOptimismPortal2(portal).depositTransaction(l2Target, 0, gasLimit, isCreation, l2Data);
+        SuperchainAddressRegistry.ChainInfo[] memory chains = superchainAddrRegistry.getChains();
+        for (uint256 i = 0; i < chains.length; i++) {
+            // Record the L1 portal call with value for action extraction.
+            IOptimismPortal2(superchainAddrRegistry.getAddress("OptimismPortalProxy", chains[i].chainId))
+                .depositTransaction(l2Target, 0, gasLimit, isCreation, l2Data);
+        }
```

Comment:
Why do we loop here?

Reply:
I understand that it should be only 1 loop cause of the config but is it right to leave it as is and not `chains[0]`?should we also add a comment or something about the purpose of the loop or is it like this in an other template?